kcidb/db/postgresql/v04_09.py

```diff
@@ -0,0 +1,165 @@
+"""Kernel CI report database - PostgreSQL schema v4.8"""
+
+import textwrap
+from kcidb.misc import merge_dicts
+import kcidb.io as io
+from kcidb.db.postgresql.schema import \
+    TextColumn, TextArrayColumn, JSONColumn, \
+    BoolColumn, TimestampColumn, Table, Index
+from .v04_08 import Schema as PreviousSchema
+
+
+CREATE_UNIT_PREFIX_TYPE_STATEMENT = textwrap.dedent("""\
+    DO $$ BEGIN
+        CREATE TYPE UNIT_PREFIX AS ENUM ('metric', 'binary');
+    EXCEPTION
+        WHEN duplicate_object THEN null;
+    END $$
+""")
+
+
+# It's OK, pylint: disable=too-many-ancestors
+class Schema(PreviousSchema):
+    """PostgreSQL database schema v4.8"""
+
+    # The schema's version.
+    version = (4, 9)
+    # The I/O schema the database schema supports
+    io = io.schema.V4_5
+
+    # A map of table names and Table constructor arguments
+    # For use by descendants
+    # Add environment_compatible, and number_* fields
+    TABLES_ARGS = merge_dicts(
+        PreviousSchema.TABLES_ARGS,
+        checkouts=merge_dicts(
+            PreviousSchema.TABLES_ARGS["checkouts"],
+            columns=dict(
+                {
+                    "git_commit_tags": TextArrayColumn(),
+                    "git_commit_message": TextColumn(),
+                    "git_repository_branch_tip": BoolColumn(),
+                }
+            )
+        )
+    )
+
+    # A map of table names and schemas
+    TABLES = {
+        name: Table(**args) for name, args in TABLES_ARGS.items()
+    }
+
+    # A map of index names and schemas
+    INDEXES = merge_dicts(PreviousSchema.INDEXES, dict(
+        checkouts_git_commit_tags=Index(
+            "checkouts", ["git_commit_tags"]
+        ),
+        checkouts_git_commit_message=Index(
+            "checkouts", ["git_commit_message"]
+        ),
```

Comment:
A regular BTREE index would be useless for this field, because nobody is going to search for the exact complete message, or even its beginning. We would need to support full-text search on this field, and for that we would need to read this chapter: https://www.postgresql.org/docs/current/textsearch.html

So let's not add the index right now. We mostly need to display the commits (or rather their subjects), right now, and that doesn't require an index.

Reply:
Dropped the index.